vm/devices/storage/disk_nvme/nvme_driver/src/driver.rs

```diff
@@ -432,6 +498,193 @@ impl<T: DeviceBacking> NvmeDriver<T> {
             .count()
     }
 
+    /// Saves the NVMe driver state during servicing.
+    pub async fn save(&mut self) -> anyhow::Result<NvmeDriverSavedState> {
+        self.nvme_keepalive = true;
+        let mut save_state = self
+            .io_issuers
+            .send
+            .call(NvmeWorkerRequest::Save, ())
+            .await?;
+
+        // Update other fields not accessible by worker task.
+        self.identify
+            .as_ref()
+            .unwrap()
+            .write_to(save_state.identify_ctrl.as_mut());
+        save_state.device_id = self.device_id.clone();
+        save_state.nsid = self.namespace.as_ref().map_or(0, |ns| ns.nsid());
+        // Either 1 or 0 namespaces per driver.
+        save_state.namespace = match &self.namespace {
+            Some(ns) => Some(ns.save()?),
+            None => None,
+        };
+        save_state.bar0_va = self.bar0_va;
+
+        Ok(save_state)
+    }
+
+    /// Attach to the backing device after restore.
+    pub async fn attach(&mut self) -> anyhow::Result<()> {
+        Ok(())
```

Comment:
What's supposed to happen here?

Reply:
There was some logic in the past, looks like the function can be removed if other parts are working as expected.